llvm/lib/Target/X86/GISel/X86InstructionSelector.cpp

```diff
@@ -195,6 +195,14 @@ X86InstructionSelector::getRegClass(LLT Ty, const RegisterBank &RB) const {
       return &X86::VR512RegClass;
   }
 
+  if (RB.getID() == X86::PSRRegBankID) {
+    if (Ty.getSizeInBits() == 80)
+      return &X86::RFP80RegClass;
+    if (Ty.getSizeInBits() == 64)
+      return &X86::RFP64RegClass;
+    return &X86::RFP32RegClass;
+  }
+
```

Comment:
I vote for the `if (Ty.getSizeInBits() ==32)` . You have an unreachable check below.